src/docs/codelabs/implicit-animations.md

```diff
@@ -0,0 +1,432 @@
+`margin` state variables: 
+
+<?code-excerpt "container{2,3}/lib/main.dart"?>
+```diff
+--- container2/lib/main.dart
++++ container3/lib/main.dart
+@@ -30,6 +30,14 @@
+     final margin = randomMargin();
+   }
+
++  void change() {
++    setState(() {
++      color = randomColor();
++      borderRadius = randomBorderRadius();
++      margin = randomMargin();
++    });
++  }
++
+   Widget build(BuildContext context) {
+     return Scaffold(
+       body: Center(
+```
+
+**4.** Now invoke the `change()` method in the `onPressed()` handler:
+
+<?code-excerpt "container{3,4}/lib/main.dart"?>
+```diff
+--- container3/lib/main.dart
++++ container4/lib/main.dart
+@@ -60,7 +60,7 @@
+                 'change',
+                 style: TextStyle(color: Colors.white),
+               ),
+-              onPressed: () => null,
++              onPressed: () => change(),
+             ),
+           ],
+         ),
+```
+
+**5.** Finally, set the duration of the animation that will power the transition 
+between the old values and the new ones:
+<?code-excerpt "container{4,5}/lib/main.dart"?>
+```diff
+--- container4/lib/main.dart
++++ container5/lib/main.dart
+@@ -2,6 +2,7 @@
+ import 'package:flutter_web_test/flutter_web_test.dart';
+ import 'package:flutter_web_ui/ui.dart' as ui;
+ import 'dart:math';
++const _duration = Duration(milliseconds: 400);
+
+ double randomBorderRadius() {
+   return Random().nextDouble() * 64;
+@@ -53,6 +54,7 @@
+                   borderRadius: BorderRadius.circular(borderRadius),
+                 ),
+               ),
++              duration: _duration,
+             ),
+             MaterialButton(
+               color: Theme.of(context).primaryColor,
+```
+
+Here’s the example with the completed changes you’ve just made – run the code
+and click on the “change” button to trigger the animation.
+
+<!-- Animated Container: https://gist.github.com/ddfbc68ec9dc28a48703d29248f5366f -->
+<iframe 
+  src="{{site.custom.dartpadx.embed-flutter-prefix}}?id=ddfbc68ec9dc28a48703d29248f5366f"
+  style="border: 1px solid lightgrey; margin-top: 10px; margin-bottom: 25px" 
+  frameborder="no" height="800" width="100%"
+></iframe>
+
+### Changing an animation curve
+
+The preceding example demonstrates animated transitions between values for 
+the container's `margin`, `borderRadius`, and `color` properties. However, 
+`AnimatedContainer` animates changes in all of its properties, including ones
+we didn't use such as `padding`, `transform`, and even `child` and `alignment`!
+
+Additionally, all implicit animations allow you to pass a [curve] parameter
+used to control how an animation changes its value over the course of the
+specified duration. In the preceding examples so far, you did not specify a 
+curve for any of the implicit animations, so the implicit animations applied a 
+[linear animation curve] by default.
+
+However, watch how the animation changes in the preceding example
+when you pass a `fastOutSlowIn` curve:
```

Comment:
Hmmmm... I'm of two minds about this. When it says "watch how the animation changes..." I'm expecting another DartPad. But, as you are probably thinking, do we NEED another DP here? I think not, but I'd rephrase this to say something like, "make the following change to the previous DartPad to see how the..."

Reply:
This was bothering me, you offered the perfect phrase for the job, many thanks Shams!

Reply:
Lol, I also was looking for another dartpad here... But yeah, telling people to modify the previous one sounds good.